### PR TITLE
Add support for json output in vtctl TopoCat command

### PR DESCRIPTION
**Pull request by dasl-**
Example:
```
vitess@localhost:/vt/src/vitess.io/vitess$ vtctl -topo_implementation zk2 -topo_global_server_address localhost:21811,localhost:21812,localhost:21813 -topo_global_root /vitess/global TopoCat -cell test -decode_proto_json 'tablets/*/Tablet'
[{"alias":{"cell":"test","uid":100},"hostname":"localhost","keyspace":"index_test","mysqlHostname":"localhost","mysqlPort":17100,"portMap":{"grpc":16100,"mysql":17100,"vt":15100},"shard":"0","type":"MASTER"},{"alias":{"cell":"test","uid":101},"hostname":"localhost","keyspace":"index_test","mysqlHostname":"localhost","mysqlPort":17101,"portMap":{"grpc":16101,"mysql":17101,"vt":15101},"shard":"0","type":"REPLICA"},{"alias":{"cell":"test","uid":200},"hostname":"localhost","keyRange":{"end":"gA=="},"keyspace":"shard_test","mysqlHostname":"localhost","mysqlPort":17200,"portMap":{"grpc":16200,"mysql":17200,"vt":15200},"shard":"-80","type":"MASTER"},{"alias":{"cell":"test","uid":201},"hostname":"localhost","keyRange":{"end":"gA=="},"keyspace":"shard_test","mysqlHostname":"localhost","mysqlPort":17201,"portMap":{"grpc":16201,"mysql":17201,"vt":15201},"shard":"-80","type":"REPLICA"},{"alias":{"cell":"test","uid":300},"hostname":"localhost","keyRange":{"start":"gA=="},"keyspace":"shard_test","mysqlHostname":"localhost","mysqlPort":17300,"portMap":{"grpc":16300,"mysql":17300,"vt":15300},"shard":"80-","type":"MASTER"},{"alias":{"cell":"test","uid":301},"hostname":"localhost","keyRange":{"start":"gA=="},"keyspace":"shard_test","mysqlHostname":"localhost","mysqlPort":17301,"portMap":{"grpc":16301,"mysql":17301,"vt":15301},"shard":"80-","type":"REPLICA"}]
```

### CODE REVIEW

File: go/vt/vtctl/topo.go [hunk 1]
```diff
@@ -106,33 +119,73 @@ func commandTopoCat(ctx context.Context, wr *wrangler.Wrangler, subFlags *flag.F
...
 	hasError := false
+	var jsonData []interface{}
 	for _, arg := range resolved {
 		data, version, err := conn.Get(ctx, arg)
 		if err != nil {
 			hasError = true
-			wr.Logger().Printf("TopoCat: Get(%v) failed: %v\n", arg, err)
+			logMsg := fmt.Sprintf("TopoCat: Get(%v) failed: %v\n", arg, err)
+			if *decodeProtoJson {
+				wr.Logger().Warningf(logMsg)
```
Comment (sougou) [inline go/vt/vtctl/topo.go]: Why should this be a warning instead of a Printf?

**Comment by dasl-** [comment]
@sougou updated to use a decoder interface. LMK what you think!

File: go/vt/vtctl/topo.go [hunk 2]
```diff
@@ -106,38 +118,17 @@ func commandTopoCat(ctx context.Context, wr *wrangler.Wrangler, subFlags *flag.F
...
-		if len(decoded) > 0 && decoded[len(decoded)-1] != '\n' && *long {
-			wr.Logger().Printf("\n")
-		}
+	var topologyDecoder TopologyDecoder
+	if *decodeProtoJson {
+		topologyDecoder = JsonTopologyDecoder{}
+	} else if *decodeProto {
+		topologyDecoder = ProtoTopologyDecoder{}
+	} else {
+		topologyDecoder = PlainTopologyDecoder{}
```
Comment (sougou) [inline go/vt/vtctl/topo.go]: Stylistic nit.

```suggestion
  switch {
  case *decodeProtoJson:
    topologyDecoder = JsonTopologyDecoder{}
  case *decodeProto:
    topologyDecoder = ProtoTopologyDecoder{}
  default:
    topologyDecoder = PlainTopologyDecoder{}
```

Comment (dasl-) [reply go/vt/vtctl/topo.go]: 👍 updated!

**Pull request merged**